### PR TITLE
cranelift: simplify `x-x` to `0`

### PR DESCRIPTION
**Pull request by Kmeakin**

### CODE REVIEW

**Review by fitzgen — APPROVED** [review]
Thanks!

Out of curiosity, is this something you've seen in the wild, or is it just for completeness?

File: cranelift/codegen/src/opts/algebraic.isle [hunk 1]
```diff
@@ -32,6 +32,9 @@
                       x))
       (ineg ty x))
 
+;; x-x == 0.
+(rule (simplify (isub ty x x)) (subsume (iconst ty (imm64 0))))
```
Comment (cfallin) [inline cranelift/codegen/src/opts/algebraic.isle:36]: This works great for all types except `$I128`, which for practical reasons (size of operator struct) isn't supported by `iconst`; so we need to guard this rule by a `fits_in_64` pattern, as for `x ^ x` on line 113 below.

**Review by cfallin — COMMENTED** [review]
Thanks! Just one nit below and then this should be good to merge.

**Comment by Kmeakin** [comment]
> Thanks!
> 
> Out of curiosity, is this something you've seen in the wild, or is it just for completeness?

Just for completeness

File: cranelift/codegen/src/opts/algebraic.isle (hunk 1 above)
Comment (fitzgen) [reply cranelift/codegen/src/opts/algebraic.isle:36]: Good catch, thanks!

**Review by cfallin — APPROVED** [review]
Thanks!

**Comment by alexcrichton** [comment]
Apologies for the merge queue frobbing here GitHub has been acting up today and I'm not really sure what's going on...

**Pull request merged**